### PR TITLE
[feature](multi catalog nereids)Add file scan node to nereids.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Add file scan node to nereids, so that the new planner could support external hms table.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.62 seconds
 load time: 671 seconds
 storage size: 17123660262 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221220084030_clickbench_pr_65541.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -491,6 +495,29 @@ public PlanFragment visitPhysicalOlapScan(PhysicalOlapScan olapScan, PlanTransla
         return planFragment;
     }
 
+    @Override
+    public PlanFragment visitPhysicalFileScan(PhysicalFileScan fileScan, PlanTranslatorContext context) {
+        // Create OlapScanNode
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:500]: ExternalFileScanNode

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 2]
```diff
@@ -63,12 +68,37 @@ public Rule build() {
...
+            throw new RuntimeException(String.format("Catalog %s does not exist.", catalogName));
+        }
+        DatabaseIf<TableIf> db = null;
+        try {
+            db = (DatabaseIf<TableIf>) catalog.getDb(dbName)
+                    .orElseThrow(() -> new RuntimeException("Database [" + dbName + "] does not exist."));
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+        db.readLock();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:93]: No need to add db lock

**Review by morningman — COMMENTED** [review]
Please add some regression test

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 3]
```diff
@@ -82,48 +112,62 @@ private LogicalPlan bindWithCurrentDb(CascadesContext cascadesContext, UnboundRe
...
+                } else {
+                    return new LogicalOlapScan(RelationUtil.newRelationId(),
+                        (OlapTable) table, ImmutableList.of(dbName));
+                }
+            case VIEW:
+                Plan viewPlan = parseAndAnalyzeView(((View) table).getDdlSql(), cascadesContext);
+                return new LogicalSubQueryAlias<>(table.getName(), viewPlan);
+            case HMS_EXTERNAL_TABLE:
+                return new LogicalFileScan(cascadesContext.getStatementContext().getNextRelationId(),
+                    (HMSExternalTable) table, ImmutableList.of(dbName));
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:167]: I suggest rename LogicalFileScan to LogicalExternalRelation(with field: HMSExternalTable table) or LogicalHMSTable

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:164]: I'll take a look at this, probably fix it in next PR.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**